src/api/app/models/group.rb

```diff
@@ -80,16 +81,13 @@ def update_from_xml(xmlhash)
     end
 
     # delete all users which were not listed
-    cache.each do |login_id, _gu|
-      GroupsUser.where('user_id = ? AND group_id = ?', login_id, id).delete_all
+    cache.each do |login_id, _|
+      delete_user(GroupsUser, login_id, id)
     end
   end
 
   def add_user(user)
-    return if users.find_by_id(user.id) # avoid double creation
-
-    gu = GroupsUser.create(user: user, group: self)
-    gu.save!
+    return if users.find_by_id(user.id) || GroupsUser.create!(user: user, group: self)
```

Comment:
Is `return if` really needed here? It could be removed unless I'm missing something.

Reply:
done!